packages/tools/accessibility/src/HtmlTwin/htmlTwinHostComponent.tsx

```diff
@@ -135,15 +135,21 @@ export class HTMLTwinHostComponent extends React.Component<IHTMLTwinHostComponen
                 scene.onNewTextureAddedObservable,
                 scene.onNewTextureAddedObservable.add((newTexture) => {
                     if (this._isFullscreenGUI(newTexture)) {
-                        const advTexture = newTexture as AdvancedDynamicTexture;
-                        setTimeout(() => {
-                            advTexture.onLoadObservable.addOnce(() => {
-                                updateA11yTree();
-                                advTexture.getChildren().forEach((control) => {
-                                    addGUIObservers(control);
+                        if (newTexture.isReady()) {
+                            updateA11yTree();
+                            newTexture.getChildren().forEach((control) => {
+                                addGUIObservers(control);
+                            });
+                        } else {
+                            setTimeout(() => {
+                                newTexture.onLoadObservable.addOnce(() => {
+                                    updateA11yTree();
+                                    newTexture.getChildren().forEach((control) => {
+                                        addGUIObservers(control);
+                                    });

```

Comment:
You should do it this way instead:
```typescript
newTexture.onLoadObservable.addOnce(() => {
    setTimeout(() => {
        updateA11yTree();
        newTexture.getChildren().forEach((control) => {
            addGUIObservers(control);
        });
    });
});
```
because technically there's a possibility with your current code that the completion of the `newTexture` build will occur in a microtask that will run before yours (the one created with `setTimeout()`). If this happens, `onLoadObservable` will be notified by this microtask, and since you haven't attached your own observer yet, you won't be notified.

Reply:
Simple solution for this - take the callback out, call it if it is loaded or register the observer if not.

Reply:
It seems like this is the "onLoaded" code:

``` javascript
updateA11yTree();
newTexture.getChildren().forEach((control) => {
   addGUIObservers(control);
});
``` 

It also happens in 139-142. take it out to the class itself, and call it wherever needed.

Reply:
Done